core/src/main/java/eu/fasten/core/plugins/KafkaPlugin.java

```diff
@@ -0,0 +1,16 @@
+package eu.fasten.core.plugins;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface KafkaPlugin<I, O> extends FastenPlugin {
+    Optional<List<String>> consumeTopics();
+
+    void setTopic(String topicName);
+
+    void consume(I record);
+
+    Optional<O> produce();
+
+    boolean recordProcessSuccessful();
```

Comment:
What is the point of this method? If the plug-in creates an error (returned through the `getPluginError()` in the main class) then this will return `false`; else it will return `true`.

Reply:
This method was used only in place on the server. I removed it as it can be easily replaced in that one place.